src/main/java/net/snowflake/client/core/SessionUtilExternalBrowser.java

```diff
@@ -227,13 +246,26 @@ void authenticate() throws SFException, SnowflakeSQLException {
       // main procedure
       int port = this.getLocalPort(ssocket);
       logger.debug("Listening localhost:{}", port);
-      String ssoUrl = getSSOUrl(port);
-      this.handlers.output(
-          "Initiating login request with your identity provider. A "
-              + "browser window should have opened for you to complete the "
-              + "login. If you can't see it, check existing browser windows, "
-              + "or your OS settings. Press CTRL+C to abort and try again...");
-      this.handlers.openBrowser(ssoUrl);
+
+      if (loginInput.getDisableConsoleLogin()) {
+        // Access GS to get SSO URL
+        String ssoUrl = getSSOUrl(port);
+        this.handlers.output(
+            "Initiating login request with your identity provider. A "
+                + "browser window should have opened for you to complete the "
+                + "login. If you can't see it, check existing browser windows, "
+                + "or your OS settings. Press CTRL+C to abort and try again...");
+        this.handlers.openBrowser(ssoUrl);
+      } else {
+        // Multiple SAML way to do authentication via console login
+        String consoleLoginUrl = getConsoleLoginUrl(port);
+        this.handlers.output(
+            "Initiating login request with your identity provider(s). A "
```

Comment:
the string is different from the one above only because of `(s)` - can we extract it to the common method that shows only this difference?

Reply:
This is a template I kept. I need to double check from DB sec team which verbiage we need to keep.